src/modules/number/index.ts

```
@@ -102,14 +102,19 @@ export class NumberModule {
     if (max < min) {
       throw new FakerError(`Max ${max} should be greater than min ${min}.`);
     }
-
-    const factor = 1 / precision;
-    const int = this.int({
-      min: min * factor,
-      max: max * factor,
-    });
-
-    return int / factor;
+    if (precision !== undefined) {
+      const factor = 1 / precision;
```

I think we need to check for `precision == 0` as otherwise this will result in `Infinity`

---

Maybe could just check precision>0?

Zero and negative precisions are meaningless.

---

Should we throw an error if precision is defined but invalid?

---

Honestly I think precision is not a very useful parameter and we should move towards deprecating it. 

For example what would a precision of 0.7 or 3 intuitively mean?

And JavaScript rounding rules means you don't necessarily get the expected number of decimal places if you pass a precision of say 0.001. 

Most use cases would be much better served by calling toFixed on the float

---

Well, precision allows for custom steps in the values.

E.g. 0.25 -> 0, 0.25, 0.5, 0.75, ...
Or 4 -> 0, 4 , 8, 12, ...

---

https://github.com/faker-js/faker/issues/1596#issuecomment-1326654346